lib/honeybadger.ex

```diff
@@ -195,48 +192,62 @@ defmodule Honeybadger do
 
   Send a notification as a `badarg` atom:
 
-      iex> Honeybadger.notify(:badarg, %{})
+      iex> Honeybadger.notify(:badarg)
       :ok
 
   If desired additional metadata can be provided as well:
 
       iex> Honeybadger.notify(%RuntimeError{}, %{culprit_id: 123})
       :ok
   """
-  @spec notify(Notice.noticeable(), map, list) :: :ok
+  @spec notify(Notice.noticeable(), map(), list()) :: :ok
   def notify(exception, metadata \\ %{}, stacktrace \\ []) do
+    backtrace = Backtrace.from_stacktrace(stacktrace)
+
     exception
-    |> Notice.new(contextual_metadata(metadata), backtrace(stacktrace))
+    |> Notice.new(contextual_metadata(metadata), backtrace)
     |> Client.send_notice()
   end
 
   @doc """
-  Retrieves the context that will get sent to the Honeybadger API when/if an
-  exception occurs in the current process.
+  Retrieves the context that will be sent to the Honeybadger API when an exception occurs in the
+  current process.
+
+  Context is stored as Logger metadata, and is in fact an alias for `Logger.metadata/0`.
   """
-  @spec context :: context
+  @spec context() :: map()
   def context do
-    Process.get(@context, %{})
+    Logger.metadata() |> Map.new()
   end
 
   @doc """
-  Merges `additional_context` into the the context that will get sent to the
-  Honeybadger API when/if an exception occurs in the current process.
+  Store additional context in the process metadata.
+
+  This function will merge the given map or keyword list into the existing metadata, with the
+  exception of setting a key to `nil`, which will remove that key from the metadata.
+
+  Context is stored as Logger metadata.
   """
-  @spec context(map | keyword) :: context
-  def context(additional_context)
-      when is_map(additional_context) or is_list(additional_context) do
-    Process.put(@context, Map.merge(context(), Enum.into(additional_context, %{})))
+  @spec context(map() | keyword()) :: map()
+  def context(map) when is_map(map), do: context(Keyword.new(map))
+
+  def context(keyword) do
+    Logger.metadata(keyword)
```

Comment:
Does setting context work like before if I'm notifying manually from the current process? I.e., if I synchronously call `Honeybadger.context(%{foo:"bar"})`, and then `Honeybadger.notify("testing")`, the context will be reported with the error (and I assume no stack trace)?

Reply:
It works just like it did before. We were using the process dictionary to store context, and that is exactly how the logger stores metadata.